src/base-input.js

```diff
@@ -84,6 +85,8 @@ export default class BaseInput extends LitElement {
       label:                   { type: String },
       name:                    { type: String },
       type:                    { type: String },
+      format:                  { type: Object },
```

Comment:
I'd argue that this is not to be in the public API? This object is only for internal use, right? Is there a use case when someone would pass that in? 

Do we need to set this property?

Reply:
This works only because it's passed in. Every example in the doc passes in a format JSON blob. In some cases it's as simple as `"{"creditCard": true}"` but it could be far more complex if the implementer wants. They can pass in any JSON object supported by CleaveJS to format for dates, phone numbers or anything else they may want.

Reply:
Sorry, I should have been more clear. Since the only format we are releasing is the credit card, I am wondering if this can be an internal setting. All the other features are TBD as to when they will be released.  

When we get to things like `format='{"date": true, "delimiter": "-", "datePattern": ["Y", "m", "d"]}` I would want to be more controlled in how this API works as there are specific formats we are to support per design and UX writing rules. https://auro.alaskaair.com/components/auro/datetime

I see the value in the open-ended API for cleave, but for our purposes, we will want this to be more specific.

Reply:
I figured out a pretty simple way to restrict the format to only apply if it's set to be a credit card. Essentially, just calling cleaveJS if the formatJSON has the `creditCard` key in it. Doing it this way means enabling other formats in the future would preserve a consistent approach to defining the format rather than credit cards having a custom wrapper layer. 

I removed the other format examples from the demo page as well.